apps/users/src/app/app.controller.ts

```diff
@@ -999,4 +1030,54 @@ export class UserController {
     const meta = createCastcleMeta(userReferrer.items, userReferrer.total);
     return { payload: response, meta: meta };
   }
+
+  @CastcleBasicAuth()
+  @ApiResponse({
+    status: 201,
+    type: ContentResponse
+  })
+  @Post(':id/recasted')
+  async recastContent(
+    @Param('id') id: string,
+    @Body('contentId') contentId: string,
+    @Req() req: CredentialRequest
+  ) {
+    const content = await this._getContentIfExist(contentId);
+    const user = await this._getUserFromBody(req, id);
+    const result = await this.contentService.recastContentFromUser(
+      content,
+      user
+    );
+
+    return this.contentService.convertContentToContentResponse(
+      user,
+      result.recastContent
+    );
+  }
+
+  @ApiResponse({
+    status: 201,
+    type: ContentResponse
+  })
+  @CastcleBasicAuth()
+  @Post(':id/quotecast')
+  async quoteContent(
+    @Param('id') id: string,
+    @Body('castcleId') contentId: string,
+    @Body('message') message: string,
+    @Req() req: CredentialRequest
+  ) {
+    const content = await this._getContentIfExist(contentId);
+
+    const user = await this._getUserFromBody(req, id);
```

Comment:
```suggestion
    @Auth() authorizer: Authorizer
  ) {
    authorizer.requestAccessForUser(id);
```

use this to check authority